tests/python/DodgyBearTest.py

```diff
@@ -0,0 +1,102 @@
+                         'AWS_SECRET_ACCESS_KEY = '
+                         '"A8+6AN5XSUZ3vysJg68Rt\A9E7duMlfKODwb3ZD8"')
+        self.assertEqual(self.get_results(), [
+                         'Amazon Web Services secret key',
+                         'Possible hardcoded secret key'])
+        os.remove('amazon.py')
+
+    def test_diff(self):
+        create_test_file('diff.py',
+                         '<<<<<<< HEAD'
+                         '\nvar = "test"\n=======\nvar = "test"\n'
+                         '>>>>>>> branch')
+        self.assertEqual(self.get_results(), [
+                         'Possible SCM diff in code',
+                         'Possible SCM diff in code'])
+        os.remove('diff.py')
+
+    def test_password(self):
+        create_test_file('passwords.py',
+                         'FACEBOOK_PASSWORD = "random@123"\n'
+                         'PASSWORD_TO_GITHUB = "random@123"\n'
+                         'PASSWORD = "random@123"\n'
+                         'PASSWORD = ""')
+        self.assertEqual(self.get_results(), [
+                         'Possible hardcoded password',
+                         'Possible hardcoded password',
+                         'Possible hardcoded password'])
+        os.remove('passwords.py')
+
+    def test_secret(self):
+        create_test_file('secrets.py',
+                         'SECRET = "random@123"\n'
+                         'SECRET_KEY = "random@123"\n'
+                         'THE_SUPER_SECRET = "random@123"')
+        self.assertEqual(self.get_results(), [
+                         'Possible hardcoded secret key',
+                         'Possible hardcoded secret key',
+                         'Possible hardcoded secret key'])
+        os.remove('secrets.py')
+
+    def test_ssh_private_key(self):
+        create_test_file('ssh_private_key.py',
+                         '-----BEGIN RSA PRIVATE KEY-----\n'
+                         'RANDOM@123\n'
+                         '-----END RSA PRIVATE KEY-----')
+        self.assertEqual(self.get_results(), [
+                         'Possible SSH private key',
+                         'Possible SSH private key'])
+        os.remove('ssh_private_key.py')
+
+    def test_ssh_public_key(self):
+        create_test_file('ssh_public_key.py',
+                         'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDMqZ2FO/Y'
+                         'y+ci0+9y3xUeonX/4tmfYPiAkjsfUWWkykHkCYUlTVS92pr'
+                         'aiTO256xy29osYk1xs7U0OpLztfHanHBuUhZz0pn/RvyYGe'
+                         'q3i5H3Ne/Z21TldMakhaKMB42V1C2+OV0xnjoEtPYUVG8ici'
+                         '7EzjLyDotRkYigRouI26PsmKPyyfqZ3TYVfMtkP21RdcI4/l'
+                         'na6d1dBmrQ3ly0B6g+pEf8ObaBriu0Fest0yZhyF8hsgueGu'
+                         'gZFXzjaAu68ib99nP17w111cbej9jUNnyfseykl5ngqrvsP3'
+                         'mGKw739z+p0vIIchHYa3jn6xW8fim/PH4gtcWC8mUXlv6XT '
+                         'random@123')
```

Comment:
I think you should use test files :) See other tests that have functions like `get_testfile_name`/`get_testfile_path` and `load_testfile`

Reply:
okay ;) :+1:

Reply:
same thing - https://github.com/coala/coala-bears/pull/2246#discussion_r169984248